src/Symfony/Component/Config/Builder/ConfigBuilderGenerator.php

```diff
@@ -170,15 +174,16 @@ private function handlePrototypedArrayNode(PrototypedArrayNode $node, ClassBuild
 
         $parameterType = $this->getParameterType($prototype);
         if (null !== $parameterType || $prototype instanceof ScalarNode) {
+            $class->addUse(ParamConfigurator::class);
             $property = $class->addProperty($node->getName());
             if (null === $key = $node->getKeyAttribute()) {
                 // This is an array of values; don't use singular name
                 $body = '
 /**
- * @param list<TYPE> $value
+ * @param ParamConfigurator|list<TYPE> $value
```

Comment:
this should be `ParamConfigurator|list<TYPE|ParamConfigurator>`, as you could use the param inside the array too

Reply:
Thank you